gtsfm/scene_optimizer.py

```diff
@@ -103,6 +104,8 @@ def create_computation_graph(
         camera_intrinsics_graph: List[Delayed],
         image_shape_graph: List[Delayed],
         gt_cameras_graph: Optional[List[Delayed]] = None,
+        gt_pose_graph: Optional[List[Delayed]] = None,
+        gt_scene_mesh: Optional[trimesh.Trimesh] = None,
```

Comment:
this is a big change to the SceneOptimizer. I really would like to see some examples of when Sampson fails and when we need this, before we check it in for all datasets (since we only have a mesh for 1 of 9 benchmark runs in the CI currently)

Reply:
There are examples on the last weeks presentation. I can add more benchmarks if you'd like.

I'm open to suggestions on how to run this evaluation without having top pass anything to `SceneOptimizer`

Reply:
Cool, I guess we didn't get to discuss those yet. Could you add them to the PR description?

Reply:
Added

Reply:
Could we add an explanation of what the technical specification of this mesh is? e.g. Is it a triangle mesh, or is it a quad mesh? Need the face vertices be arranged in CCW or CW order, etc?

Reply:
I believe I mention that it's a triangular surface mesh in multiple places, and I think the fact that the library is called `trimesh` gives it away. 

There are many different mesh formats that `trimesh` can handle, so I don't think it would be useful to include specific formatting details. If your mesh was generated properly for the given format (e.g., PLY, OBJ, STL), `trimesh` can handle it.